kratos/includes/constitutive_law.h

```diff
@@ -1165,7 +1165,7 @@ class KRATOS_API(KRATOS_CORE) ConstitutiveLaw : public Flags
      * @see Parameters
      * @see StressMeasures
      */
-    void FinalizeMaterialResponse (Parameters& rValues,const StressMeasure& rStressMeasure);
+    virtual void FinalizeMaterialResponse (Parameters& rValues,const StressMeasure& rStressMeasure);
```

Comment:
A priori this is already virtual in their different calls:

~~~c++
    /**
     * Finalize the material response in terms of 1st Piola-Kirchhoff stresses
     * @see Parameters
     */

    virtual void FinalizeMaterialResponsePK1 (Parameters& rValues);

    /**
     * Finalize the material response in terms of 2nd Piola-Kirchhoff stresses
     * @see Parameters
     */

    virtual void FinalizeMaterialResponsePK2 (Parameters& rValues);

    /**
     * Finalize the material response in terms of Kirchhoff stresses
     * @see Parameters
     */

    virtual void FinalizeMaterialResponseKirchhoff (Parameters& rValues);

    /**
     * Finalize the material response in terms of Cauchy stresses
     * @see Parameters
     */

    virtual void FinalizeMaterialResponseCauchy (Parameters& rValues);
~~~
Are you addding an additional measure?

Reply:
Opinion @AlejandroCornejo ?

Reply:
well se should check the effect of this elimination of virtual. I think that this method is never overriden but who knows... If it's never overriden could be a good thing this change

Reply:
The point is why override, you should override the submethods, except if you have another stress measures, but then maybe you should propose the new stress measure instead of this.

Reply:
Agree with @loumalouomega observation. This must remain as non-virtual. The ones to be overridden are the corresponding stress measures.

Reply:
ah! reviewing it fast I though he that was removing the virtual. If it is adding the virtual I would avoid that unnecessary overhead.

Reply:
Thanks for the quick reply / review I will do the necessary separation tomorrow and remove the virtual.

Reply:
Reverted